File: android/src/main/kotlin/com/tieorange/image_cache_plugin/domain/CachedImageFile.kt
package com.tieorange.image_cache_plugin.domain

import java.io.File

/** Origin of a cached image returned by the repository. */
enum class CacheSource { NETWORK, DISK }

/** Immutable metadata for a complete cached image file. */
data class CachedImageFile(
    val file: File,
    val source: CacheSource,
    val cachedAtMilliseconds: Long,
)

File: lib/src/cached_image_file.dart
/// Identifies where a cached image was obtained.
enum ImageCacheSource { network, disk }

/// A native cached image file and its cache metadata.
final class CachedImageFile {
  /// Creates cache metadata for [path].
  const CachedImageFile({
    required this.path,
    required this.source,
    required this.cachedAtMilliseconds,
  });

  /// Absolute path to the native cache file.
  final String path;

  /// Whether this request downloaded the file or found it on disk.
  final ImageCacheSource source;

  /// Successful native commit time in epoch milliseconds.
  final int cachedAtMilliseconds;
}
